### ISSUE TITLE
3.0.2 Installer Error during setupDatabase()

### ISSUE THREAD

**Issue by LuMAXW3B** [croogo/croogo#850]
Hi @rchavik 

How did you bake the seed files for phinx migrations? - There is a migrations issue at the moment during the setupDatabase() method. The locales and settings seed make some noise.

Can you provide me your default sql dump, then I would bake the correct seed files.

Here my stack trace from the error.log:

2017-12-12 13:51:48 Error: [PDOException] SQLSTATE[HY000]: General error: 1366 Incorrect string value: '\xF0\x91\x84\x9D\xF0\x91...' for column 'native' at row 1
Request URL: /install/data
Referer URL: http://localhost/croogo/install/database
Stack Trace:
#0 /Users/lukasmarks/Sites/croogo/vendor/robmorgan/phinx/src/Phinx/Db/Adapter/PdoAdapter.php(392): PDOStatement->execute(Array)
#1 /Users/lukasmarks/Sites/croogo/vendor/cakephp/migrations/src/CakeAdapter.php(400): Phinx\Db\Adapter\PdoAdapter->insert(Object(Phinx\Db\Table), Array)
#2 /Users/lukasmarks/Sites/croogo/vendor/robmorgan/phinx/src/Phinx/Db/Table.php(651): Migrations\CakeAdapter->insert(Object(Phinx\Db\Table), Array)
#3 /Users/lukasmarks/Sites/croogo/vendor/robmorgan/phinx/src/Phinx/Db/Table.php(639): Phinx\Db\Table->saveData()
#4 /Users/lukasmarks/Sites/croogo/vendor/robmorgan/phinx/src/Phinx/Db/Table.php(675): Phinx\Db\Table->update()
#5 /Users/lukasmarks/Sites/croogo/vendor/croogo/croogo/Settings/config/Seeds/LanguagesSeed.php(61): Phinx\Db\Table->save()
#6 /Users/lukasmarks/Sites/croogo/vendor/robmorgan/phinx/src/Phinx/Migration/Manager/Environment.php(154): LanguagesSeed->run()
#7 /Users/lukasmarks/Sites/croogo/vendor/robmorgan/phinx/src/Phinx/Migration/Manager.php(395): Phinx\Migration\Manager\Environment->executeSeed(Object(LanguagesSeed))
#8 /Users/lukasmarks/Sites/croogo/vendor/robmorgan/phinx/src/Phinx/Migration/Manager.php(514): Phinx\Migration\Manager->executeSeed('default', Object(LanguagesSeed))
#9 [internal function]: Phinx\Migration\Manager->seed('default', NULL)
#10 /Users/lukasmarks/Sites/croogo/vendor/cakephp/migrations/src/Migrations.php(319): call_user_func_array(Array, Array)
#11 /Users/lukasmarks/Sites/croogo/vendor/cakephp/migrations/src/Migrations.php(269): Migrations\Migrations->run('seed', Array, Object(Symfony\Component\Console\Input\ArrayInput))
#12 /Users/lukasmarks/Sites/croogo/vendor/croogo/croogo/Core/src/Plugin.php(562): Migrations\Migrations->seed(Array)
#13 /Users/lukasmarks/Sites/croogo/vendor/croogo/croogo/Install/src/InstallManager.php(212): Croogo\Core\Plugin->seed('Croogo/Settings')
#14 /Users/lukasmarks/Sites/croogo/vendor/croogo/croogo/Install/src/InstallManager.php(162): Croogo\Install\InstallManager->seedTables('Croogo/Settings')
#15 /Users/lukasmarks/Sites/croogo/vendor/croogo/croogo/Install/src/Controller/InstallController.php(192): Croogo\Install\InstallManager->setupDatabase()
#16 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Controller/Controller.php(440): Croogo\Install\Controller\InstallController->data()
#17 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/ActionDispatcher.php(119): Cake\Controller\Controller->invokeAction()
#18 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/ActionDispatcher.php(93): Cake\Http\ActionDispatcher->_invoke(Object(Croogo\Install\Controller\InstallController))
#19 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/BaseApplication.php(108): Cake\Http\ActionDispatcher->dispatch(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response))
#20 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/Runner.php(65): Cake\Http\BaseApplication->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response), Object(Cake\Http\Runner))
#21 /Users/lukasmarks/Sites/croogo/vendor/croogo/croogo/Install/src/Middleware/InstallMiddleware.php(24): Cake\Http\Runner->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response))
#22 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/Runner.php(65): Croogo\Install\Middleware\InstallMiddleware->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response), Object(Cake\Http\Runner))
#23 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Routing/Middleware/RoutingMiddleware.php(104): Cake\Http\Runner->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response))
#24 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/Runner.php(65): Cake\Routing\Middleware\RoutingMiddleware->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response), Object(Cake\Http\Runner))
#25 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Routing/Middleware/AssetMiddleware.php(88): Cake\Http\Runner->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response))
#26 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/Runner.php(65): Cake\Routing\Middleware\AssetMiddleware->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response), Object(Cake\Http\Runner))
#27 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Error/Middleware/ErrorHandlerMiddleware.php(96): Cake\Http\Runner->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response))
#28 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/Runner.php(65): Cake\Error\Middleware\ErrorHandlerMiddleware->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response), Object(Cake\Http\Runner))
#29 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/Runner.php(51): Cake\Http\Runner->__invoke(Object(Cake\Http\ServerRequest), Object(Cake\Http\Response))
#30 /Users/lukasmarks/Sites/croogo/vendor/cakephp/cakephp/src/Http/Server.php(81): Cake\Http\Runner->run(Object(Cake\Http\MiddlewareQueue), Object(Cake\Http\ServerRequest), Object(Cake\Http\Response))
#31 /Users/lukasmarks/Sites/croogo/webroot/index.php(40): Cake\Http\Server->run()
#32 {main}

2017-12-12 13:51:48 Warning: The connection is going to be closed but there is an active transaction.

**Comment by LuMAXW3B** [comment]
FYI, the same error while shell installer, there is a seed data issue.

**Comment by rchavik** [comment]
Try using utf8mb4 charset with utf8mb4_unicode_ci collation

**Comment by LuMAXW3B** [comment]
@rchavik utf8mb4_unicode_ci is the DB encoding I use in general since CakePHP 3.

FYI; Machine: Mac OSX with latest native compiled LAMP on brew stack. Server-Version: 5.7.20 - Homebrew
I suppose the baked Seed has a encoding issue.

**Comment by rchavik** [comment]
weird.  the seed in question is actually dynamically generated by php:

 https://github.com/croogo/croogo/blob/master/Settings/config/Seeds/LanguagesSeed.php#L38-L59

maybe some issue with php 5.7 resource bundle

**Comment by LuMAXW3B** [comment]
@rchavik its MYSQL Server-Version: 5.7.20 - Homebrew running on PHP-Version: 7.0.26.
I will test the Installer with default utf8_unicode_ci - when it works maby we have to define the encoding...

http://docs.phinx.org/en/latest/migrations.html?highlight=encoding

**Comment by rchavik** [comment]
haha, i made up php 5.7.

**Comment by rchavik** [comment]
fyi, it's suppose to load something like:

![image](https://user-images.githubusercontent.com/39490/33896814-6270cf94-df96-11e7-8418-405c63e107d0.png)

see it [here](http://demo.croogo.org/3.0/admin/settings/languages)

**Comment by LuMAXW3B** [comment]
@rchavik I think I found one problem:

![grafik](https://user-images.githubusercontent.com/1556376/33897353-7a41a0fc-df65-11e7-99ae-a429752dbb5b.png)

Why is the migration cretaing a utf8_general_ci table encoding...?

**Comment by rchavik** [comment]
i think that's phinx's doing.

**Comment by LuMAXW3B** [comment]
I think we should Re-Bake the migrations and seeds based on your latest MySQL DB and Table encoding.

> i think that's phinx's doing.
Exactly and thats definitely wrong

**Comment by LuMAXW3B** [comment]
When you provide me the Croogo 3 Demo SQL Dump it will work on my machine. Its all about some Phinx migrations + seed issue here

**Comment by rchavik** [comment]
can you share output of:

```
mysql> show create database croogo3;
+----------+------------------------------------------------------------------------------------------------+
| Database | Create Database                                                                                |
+----------+------------------------------------------------------------------------------------------------+
| croogo3  | CREATE DATABASE `croogo3` /*!40100 DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci */ |
+----------+------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> show full columns from languages;
+------------+--------------+--------------------+------+-----+---------+----------------+---------------------------------+---------+
| Field      | Type         | Collation          | Null | Key | Default | Extra          | Privileges                      | Comment |
+------------+--------------+--------------------+------+-----+---------+----------------+---------------------------------+---------+
| id         | int(11)      | NULL               | NO   | PRI | NULL    | auto_increment | select,insert,update,references |         |
| title      | varchar(255) | utf8mb4_unicode_ci | NO   |     | NULL    |                | select,insert,update,references |         |
| native     | varchar(255) | utf8mb4_unicode_ci | NO   |     | NULL    |                | select,insert,update,references |         |
| alias      | varchar(255) | utf8mb4_unicode_ci | NO   |     | NULL    |                | select,insert,update,references |         |
| locale     | varchar(15)  | utf8mb4_unicode_ci | NO   | UNI | NULL    |                | select,insert,update,references |         |
| status     | tinyint(1)   | NULL               | NO   |     | 1       |                | select,insert,update,references |         |
| weight     | int(11)      | NULL               | YES  |     | NULL    |                | select,insert,update,references |         |
| updated    | datetime     | NULL               | YES  |     | NULL    |                | select,insert,update,references |         |
| updated_by | int(20)      | NULL               | YES  |     | NULL    |                | select,insert,update,references |         |
| created    | datetime     | NULL               | YES  |     | NULL    |                | select,insert,update,references |         |
| created_by | int(20)      | NULL               | YES  |     | NULL    |                | select,insert,update,references |         |
+------------+--------------+--------------------+------+-----+---------+----------------+---------------------------------+---------+
11 rows in set (0.00 sec)```

**Comment by LuMAXW3B** [comment]
@rchavik here you go

![grafik](https://user-images.githubusercontent.com/1556376/33898263-31581e5e-df68-11e7-96b2-cfb5df23686c.png)

**Comment by rchavik** [comment]
looks exactly the same. hmm.

maybe delete the LanguagesSeed.php just to avoid this during install for now.

**Comment by maiconpinto** [comment]
I had the same error, and the solution was drop database, and create again with:
```
CREATE DATABASE croogo
  DEFAULT CHARACTER SET utf8
  DEFAULT COLLATE utf8_general_ci;
```

**Comment by rchavik** [comment]
Try this diff:

```diff
diff --git a/Menus/config/Migrations/20160807105149_MenusInitialMigration.php b/Menus/config/Migrations/20160807105149_MenusInitialMigration.php
index 217d010..7564ee8 100644
--- a/Menus/config/Migrations/20160807105149_MenusInitialMigration.php
+++ b/Menus/config/Migrations/20160807105149_MenusInitialMigration.php
@@ -179,7 +179,7 @@ class MenusInitialMigration extends AbstractMigration
                 [
                     'alias',
                 ],
-                ['unique' => true]
+                ['unique' => true, 'limit' => 100]
             )
             ->create();
     }
diff --git a/Taxonomy/config/Migrations/20160807105245_TaxonomyInitialMigration.php b/Taxonomy/config/Migrations/20160807105245_TaxonomyInitialMigration.php
index 89108b2..0309276 100644
--- a/Taxonomy/config/Migrations/20160807105245_TaxonomyInitialMigration.php
+++ b/Taxonomy/config/Migrations/20160807105245_TaxonomyInitialMigration.php
@@ -73,7 +73,7 @@ class TaxonomyInitialMigration extends AbstractMigration
                 [
                     'slug',
                 ],
-                ['unique' => true]
+                ['unique' => true, 'limit' => 100]
             )
             ->create();
 
@@ -162,7 +162,7 @@ class TaxonomyInitialMigration extends AbstractMigration
                 [
                     'alias',
                 ],
-                ['unique' => true]
+                ['unique' => true, 'limit' => 100]
             )
             ->create();
 
@@ -231,7 +231,7 @@ class TaxonomyInitialMigration extends AbstractMigration
                 [
                     'alias',
                 ],
-                ['unique' => true]
+                ['unique' => true, 'limit' => 100]
             )
             ->create();
 
```

**Comment by rchavik** [comment]
Should be fixed in 3.0.6
